packages/components/src/snackbar/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import { flow, noop } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { RawHTML, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '../';
+
+const stopPropagation = ( event ) => event.stopPropagation();
+const NOTICE_TIMEOUT = 10000;
+
+function Snackbar( {
+	className,
+	status,
+	children,
+	actions = [],
+	__unstableHTML,
+	onRemove = noop,
+} ) {
+	useEffect( () => {
+		// This rule doesn't account yet for React Hooks
+		// eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+		const timeoutHandle = setTimeout( () => {
+			onRemove();
+		}, NOTICE_TIMEOUT );
+
+		return () => clearTimeout( timeoutHandle );
+	}, [] );
```

Comment:
shouldn't `onRemove` be included here as a dependency?

Reply:
Probably. Obviously I don't expect the `onRemove` prop to change over time. So the question here (and I had this in other components too is). Should we add the dependency here which could have a small performance cost (probably very small) or just don't support updating the prop?

Reply:
From the react docs and what Dan Abramov has written, they've been pretty explicit in suggesting dependencies should always be included.  As for when dependencies are _known_ to not change it doesn't appear it adds much overhead.  In referencing, an example he gave with a `useEffect` listing `dispatch` (from `useReducer`) as a dependency he wrote:

> (You may omit dispatch, setState, and useRef container values from the deps because React guarantees them to be static. But it also doesn’t hurt to specify them.)

Can  you guarantee that `onRemove` will never change over time?

Reply:
I guess if it doesn't hurt :) We should add it. The question becomes, should we reset the timeout once the prop changes because this will be the default behavior if I add it there :).

Reply:
well the timeout should still apply for the existing SnackNotice and it would just get reset for whatever new SnackNotice (with the changed prop) is rendered in right?  I don't think we need to worry about the reset of the timeout.

Reply:
If you add a dependency `onRemove`, the effect will trigger once the prop changes, it will clear the old timeout and trigger a new one which would reset the timeout.

Reply:
So to solve this properly we need two effects and a state :) 

 - the timeout would change a state value (shouldRemove)
 - Another effect would trigger onRemove if  "shouldRemove" is true

I'm starting to wonder if it's worth the complexity just to follow guidelines

Reply:
I don't think something like this is too complex:

```js
useEffect( () => {
	// This rule doesn't account yet for React Hooks
	// eslint-disable-next-line @wordpress/react-no-unsafe-timeout
	const timeoutHandle = setTimeout( () => {
		onRemove();
	}, NOTICE_TIMEOUT );

	return () => {
            clearTimeout( timeoutHandle );
            onRemove()
        }
}, [onRemove] );
```

useEffect captures the props and state of the render call that defined it so.

Reply:
That's not the behavior we want. In your example, there are two issues:

 - If `onRemove` changes even before the timeout ends, the notice is removed (the previous `onRemove` is called)
 - if `onRemove` changes, we setup a new timeout to clear the notice after `NOTICE_TIMEOUT`ms

What we want is to clear the notice `NOTICE_TIMEOUT` after it was mounted regardless of the value of `onRemove` prop.

Reply:
You're right.  I took some time to simulate what happens in various scenarios:

- with no dependency and `onRemove` prop changing
- with no dependency and `onRemove` prop not changing
- with `onRemove` as a dependency, the prop changing.
- with `onRemove` as a dependency, the prop not changing.
- with `onRemove` as a dependency, the prop changing and onRemove called with clearTimeout.

In each of the above scenarios I added an error notice and then a success notice to watch what happened it became more apparent why you set things up this way, so ya trying to to follow hooks rules here will make things more complex.

It would be good to have some tests to verify that notices are added and removed in the expected order.